client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-credentials/hooks/use-credentials-form.ts

```diff
@@ -60,68 +64,44 @@ export const useCredentialsForm = ( onSubmit: ( siteInfo?: UrlData ) => void ) =
 		errors: serverSideError,
 	} );
 
+	const accessMethod = watch( 'migrationType' );
+
 	useEffect( () => {
 		setIsBusy( isSubmitting );
 	}, [ isSubmitting ] );
 
-	const isWpcom = useMemo( () => !! siteInfo?.platform_data?.is_wpcom, [ siteInfo ] );
+	const isLoginFailed =
+		error?.code === 'automated_migration_tools_login_and_get_cookies_test_failed';
+	const canBypassVerification = isLoginFailed || isWPCOM( siteInfo ) || isNotWordPress( siteInfo );
+	const shouldAnalyzeUrl = ! isLoginFailed && accessMethod === 'credentials';
 
-	const canBypassVerification = useMemo( () => {
-		const credentialsVerificationFailed =
-			error?.code === 'automated_migration_tools_login_and_get_cookies_test_failed';
-		return credentialsVerificationFailed || isWpcom;
-	}, [ error, isWpcom ] );
+	const submitHandler = handleSubmit( async ( data: CredentialsFormData ) => {
+		clearErrors();
 
-	const submitHandler = useCallback(
-		async ( data: CredentialsFormData ) => {
-			clearErrors();
+		const siteInfoResult = shouldAnalyzeUrl ? await analyzeUrl( data.from_url ) : siteInfo;
+		setSiteInfo( siteInfoResult );
 
-			const payload = {
-				...data,
-				bypassVerification: canBypassVerification || ! isEnglishLocale,
-			};
-
-			const siteInfoResult = canBypassVerification ? siteInfo : await analyzeUrl( data.from_url );
-			setSiteInfo( siteInfoResult );
-
-			if ( siteInfoResult?.platform_data?.is_wpcom && ! canBypassVerification ) {
-				return;
-			}
-
-			try {
-				await requestAutomatedMigration( payload );
-			} catch ( error ) {
-				// Do nothing, error is handled by the form.
-			}
-		},
-		[
-			requestAutomatedMigration,
-			canBypassVerification,
-			isEnglishLocale,
-			clearErrors,
-			analyzeUrl,
-			siteInfo,
-		]
-	);
-
-	const getContinueButtonText = useCallback( () => {
-		if ( isEnglishLocale && isSubmitting && ! canBypassVerification ) {
-			return translate( 'Verifying credentials' );
+		if ( isWPCOM( siteInfoResult ) && ! canBypassVerification ) {
+			return;
 		}
-		if ( isEnglishLocale && canBypassVerification ) {
-			return translate( 'Continue anyways' );
+
+		if ( isNotWordPress( siteInfoResult ) ) {
+			onSubmit( siteInfoResult );
+			return;
 		}
-		return translate( 'Continue' );
-	}, [ isSubmitting, canBypassVerification, isEnglishLocale, translate ] );
 
-	useEffect( () => {
```

Comment:
The hooks is not necessary, we can capture the success directly on the submit handler.